lib/Dispatcher.js

```diff
@@ -39,29 +47,35 @@ module.exports = function () {
      * @throws {Error} if store is invalid
      * @throws {Error} if store is already registered
      */
-    Dispatcher.registerStore = function registerStore(store) {
-        if ('function' !== typeof store) {
+    registerStore: function registerStore(store) {
+        if (!BaseStore.isPrototypeOf(store)) {
```

Comment:
but their classes do need to conform to the store the same store interface or they will get some confusing error somewhere else, for instance `Dispatcher.registerStore(String)`,  `Dispatcher.registerStore(function () {})` this will not raise the expected error because those are functions but it will crash somewhere else for sure.